dlp/deid_test.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import pytest
+
+import deid
+
+harmful_string = 'My SSN is 372819127'
+parent = os.environ['GCLOUD_PROJECT']
+wrapped_key = os.environ['DLP_DEID_WRAPPED_KEY']
+key_name = os.environ['DLP_DEID_KEY_NAME']
+surrogate_type = 'SSN_TOKEN'
+csv_file = 'resources/dates.csv'
+output_csv_file = 'resources/temp.results.csv'
+date_shifted_amount = 30
+date_fields = ['birth_date', 'register_date']
+csv_context_field = 'name'
+
+
+# [START Deidentify with masking]
+def test_deidentify_with_mask(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is *********' in out
+
+
+def test_deidentify_with_mask_ignore_insensitive_data(capsys):
+    harmless_string = 'My favorite color is blue'
+
+    deid.deidentify_with_mask(parent, harmless_string)
+
+    out, _ = capsys.readouterr()
+    assert harmless_string in out
+
+
+def test_deidentify_with_mask_masking_character_specified(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string, masking_character='#')
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is #########' in out
+
+
+def test_deidentify_with_mask_masking_number_specified(capsys):
+    harmful_string = 'My SSN is 372819127'
+
+    deid.deidentify_with_mask(parent, harmful_string, number_to_mask=7)
+
+    out, _ = capsys.readouterr()
+    assert 'My SSN is *******27' in out
+
+
+def test_deidentify_with_mask_handles_masking_number_error(capsys):
```

Comment:
We can go ahead and remove this test; it's not a problem for the sample if the API rejects this.

Reply:
done